segment_reader_test.go

```diff
@@ -0,0 +1,98 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafka
+
+import (
+	"testing"
+
+	"github.com/matryer/is"
+	"github.com/segmentio/kafka-go/sasl/plain"
+)
+
+func TestSegmentReader_TLSOnly(t *testing.T) {
+	is := is.New(t)
+
+	caCert := "Bag Attributes\n    friendlyName: caroot\n    2.16.840.1.113894.746875.1.1: <Unsupported tag 6>\nsubject=/CN=localhost/OU=CIA/O=REA/L=Melbourne/C=AU\nissuer=/CN=localhost/OU=CIA/O=REA/L=Melbourne/C=AU\n-----BEGIN CERTIFICATE-----\nMIIDHjCCAgYCCQC9iilqJUAoxzANBgkqhkiG9w0BAQsFADBRMRIwEAYDVQQDDAls\nb2NhbGhvc3QxDDAKBgNVBAsMA0NJQTEMMAoGA1UECgwDUkVBMRIwEAYDVQQHDAlN\nZWxib3VybmUxCzAJBgNVBAYTAkFVMB4XDTIyMDMyMTEyNDcyNloXDTQ5MDgwNTEy\nNDcyNlowUTESMBAGA1UEAwwJbG9jYWxob3N0MQwwCgYDVQQLDANDSUExDDAKBgNV\nBAoMA1JFQTESMBAGA1UEBwwJTWVsYm91cm5lMQswCQYDVQQGEwJBVTCCASIwDQYJ\nKoZIhvcNAQEBBQADggEPADCCAQoCggEBAK8eUXQWIEvqH6gWg9CyU1FZ9I5ZfeSz\n5BgEwQelG3YRiBmN4MXQmVErvy8JEdC9AbDdNvwsWlBD1xoWC0S2Q2qMhF6M03ny\nrrx0OwKNxdNwZvrMCin6adVS66x4R83X/YprZiS0fMtZHnrPsEVZxw7QSObGPnUV\nqinVqZh4Mo2N7tbxYa6ZALXgDf0yXbzGOGENuEaw9+5H01+6wDAwoxmm3pgQ0bF1\nyrqGh6P5ePtbaI6C+WBW/u0HgXUgyJaQA3vZIS6cOnwf76osPkFiCp5LtjTWblBd\nBwEmjtMu4n6/QEsUNM93lP/iJ7y8LWGrMFxL1700KFWlkJ8CIbpdjH0CAwEAATAN\nBgkqhkiG9w0BAQsFAAOCAQEAAJXIy4onOdKSceG3gjJHjCIZf74/Ka7rIhCnVyn1\n/EiTub70Df1b2vsPl25axv3ujMETubUzSzyyRSQ/5o61T5nZXPmmbTtU/7f1lLWk\n6tiHyKRCPd6InLnGFqhjmKp052LCwX0ZqUy2x+/uXxrYI7+wcB9QpDQsw4nnfDhb\nkym8hzUIu6IST1TbHFD7NoTA1L/qVlVQ8Sj6XVGQKmwBPxX/i1wHFTDrnS4uskvK\nFkPUsd6OmYbguwS3Ktj40C/pV3Z5OS/kR4+pO349I+b42ImzWxpgMRSVuI4y0Lvk\nm6GdbnJPvzqZT5yZmv05j6LQXkm5ugqPmOARMKrSrWACUA==\n-----END CERTIFICATE-----\n"
```

Comment:
We could use backticks to create a multiline string.
```suggestion
	caCert := `Bag Attributes
    friendlyName: caroot
...`
```

And we could reuse these certs in reader and writer tests.

Reply:
Actually, what do you think about this: https://github.com/ConduitIO/conduit-connector-kafka/pull/12/commits/07b4a58b2645f7d90985022ba2a07c5c69bc4772? Even this option, which is prettier, consumes a lot of space (visually), so I thought it's maybe good to just offload those into a file (which they are anyway).

Reply:
That's the way to go!